http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -58,72 +65,50 @@
         }
     }
 
-    private final LoginContext loginContext;
-    private final GSSCredential clientCredential;
+    private final File keytab;
+    private final File credentialCache;
     private final String remoteServiceName;
+    private final String principal;
+
+    private final LoadingCache<String, LoginContext> loginContextCache = CacheBuilder.newBuilder()
+            .expireAfterWrite(CACHE_EXPIRE_TIME.roundTo(TimeUnit.HOURS), TimeUnit.HOURS)
+            .removalListener(notification -> {
+                try {
+                    ((LoginContext)notification.getValue()).logout();
+                }
+                catch (LoginException e) {
+                    Throwables.propagate(e);
```

Comment:
Where does this exception go? Probably should log a warning (or do nothing -- what would an error mean?)